File: src/main/java/com/avaya/wiki/domain/Ebook.java
package com.avaya.wiki.domain;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Ebook {
    private Long id;
    private String name;
    private Long cat1Id;
    private Long cat2Id;
    private String descText;
    private String coverUrl;
    private String docUrl;
    private String docType;
    private String version;
    private Integer views;
    private Integer likes;
}

File: web/src/types/ebook.ts
// types/ebook.ts

// ============================================================
// General type definition for Ebook entries
// ============================================================
export interface Ebook {
    id: number
    name: string
    cat1Id: number
    cat2Id: number
    descText: string
    coverUrl: string
    docUrl: string
    docType: string
    version: string
    views: number
    likes: number
}

export interface EbookQueryParams {
    page: number
    size: number
    name?: string
}